### PR TITLE
Add custom markdown (RMarkdown) snippets

### PR DESCRIPTION
**Pull request by kevinushey**
This PR will encompass our own custom Markdown snippets. The set of snippets we want to include is still a TODO here.

### CODE REVIEW

File: src/gwt/acesupport/snippets/snippets/markdown.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   {
+      name: "![",
+      content: '![${1:label}](${2:location})'
+   },
+   {
+      name: "r",
+      content: "```{r ${1:label}, ${2:options}}\n${0}\n```"
+   },
+   {
+      name: "rcpp",
```
Comment (jjallaire) [inline src/gwt/acesupport/snippets/snippets/markdown.js]: Note that the {Rcpp} syntax isn't in knitr yet so we should probably use the old engine= syntax for now

**Comment by jjallaire** [comment]
Do we also need to explicitly add this to the Edit Snippets dialog? (or does it somehow show up there automatically)

**Comment by kevinushey** [comment]
Should be good to go now -- shall we merge to master (and add more snippets later if desired)?

**Comment by jjallaire** [comment]
Yup, I'll merge and ask Yihui to make some additional suggestions.

**Pull request merged**